### PR TITLE
Add first version of filter buttons implemented with react

### PR DESCRIPTION
**Pull request by mikkomustikkamaa**
Updated PR to contain styles and unit tests. Also refactoring is done as goal to better follow Oskari practices.

### CODE REVIEW

File: bundles/framework/layerlist/view/LayerFilters/LayerFilter/LayerFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    );
+};
+
+LayerFilter.propTypes = {
+    text: PropTypes.string.isRequired,
+    tooltip: PropTypes.string.isRequired,
+    filterName: PropTypes.string.isRequired,
+    currentStyle: PropTypes.string.isRequired,
+    clickHandler: PropTypes.func.isRequired
+};
```
Comment (jheijari) [inline bundles/framework/layerlist/view/LayerFilters/LayerFilter/LayerFilter.jsx]: Move this to bundles/framework/layerlist/view/LayerFilters/LayerFilter.jsx

File: bundles/framework/layerlist/view/LayerFilters/LayerFilters.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <StyledFilters>
+            {mappedFilters}
+        </StyledFilters>
+    );
+};
+
+LayerFilters.propTypes = {
+    filters: PropTypes.object.isRequired,
+    service: PropTypes.object.isRequired
+};
```
Comment (jheijari) [inline bundles/framework/layerlist/view/LayerFilters/LayerFilters.jsx]: Move to bundles/framework/layerlist/view/LayerFilter.jsx

File: bundles/framework/layerlist/view/LayersTab.jsx [hunk 3]
```diff
@@ -253,6 +257,83 @@ Oskari.clazz.define(
                 me._relatedKeywordsPopup(keyword, event, me);
             }
         },
+        /**
```
Comment (jheijari) [inline bundles/framework/layerlist/view/LayersTab.jsx]: Remove this dead code, please.

File: bundles/framework/layerselector2/service/layerlist.js [hunk 4]
```diff
@@ -23,6 +24,32 @@
...
+                                    o.cls.current = o.cls.deactive;
+                                    return o;
+                                });
+                            }
+                        } else {
+                            Object.values(me.layerlistFilterButtons).map(o => {
+                                o.cls.current = o.cls.deactive;
+                                return o;
+                            });
+                        }
```
Comment (jheijari) [inline bundles/framework/layerselector2/service/layerlist.js]: Create a new array for layerlistFilterButtons to enable shallow comparisation.

File: bundles/framework/layerselector2/service/layerlist.js [hunk 5]
```diff
@@ -23,6 +24,32 @@
             getName: function () {
                 return this.__name;
             },
+            getMutator () {
+                const me = this;
+                return {
```
Comment (jheijari) [inline bundles/framework/layerselector2/service/layerlist.js]: Create reusable mutator.

File: bundles/framework/layerlist/view/LayerFilters/LayerFilter/LayerFilter.jsx (hunk 1 above)
Comment (mikkomustikkamaa) [reply bundles/framework/layerlist/view/LayerFilters/LayerFilter/LayerFilter.jsx]: Changed as suggested.

File: bundles/framework/layerlist/view/LayerFilters/LayerFilters.jsx (hunk 2 above)
Comment (mikkomustikkamaa) [reply bundles/framework/layerlist/view/LayerFilters/LayerFilters.jsx]: Changed as suggested.

File: bundles/framework/layerlist/view/LayersTab.jsx (hunk 3 above)
Comment (mikkomustikkamaa) [reply bundles/framework/layerlist/view/LayersTab.jsx]: Changed as suggested.

File: bundles/framework/layerselector2/service/layerlist.js (hunk 4 above)
Comment (mikkomustikkamaa) [reply bundles/framework/layerselector2/service/layerlist.js]: Changed as suggested.

File: bundles/framework/layerselector2/service/layerlist.js [hunk 6]
```diff
@@ -3,6 +3,34 @@
         function () {
             this.layerlistFilterButtons = {};
             this.currentFilter = null;
+            const me = this;
+            this.mutator = {
```
Comment (jheijari) [inline bundles/framework/layerselector2/service/layerlist.js]: Move mutator creation to it's own function, away from the constructor function.

Comment (jheijari) [reply bundles/framework/layerselector2/service/layerlist.js]: Otherwise looks good for merging

Comment (mikkomustikkamaa) [reply bundles/framework/layerselector2/service/layerlist.js]: Changed as suggested.

**Pull request merged**